# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/script/ScriptLoader.cpp
# Commit: 02269004c755
# Full Hash: 02269004c7554e6972c13ee6d147fc2ddcee34f7
# Author: Yoshi Cheng-Hao Huang <allstars.chh@gmail.com>
# Date: 2025-09-26 06:07 +0000
# Regressor Bug: 1968890
# File Overlap Count: 5
# Overlapping Files: js/loader/ModuleLoadRequest.cpp, js/loader/ModuleLoaderBase.cpp, js/loader/ModuleLoadRequest.h, js/loader/ModuleLoaderBase.h, dom/script/ScriptLoader.cpp
# Description:
#   Bug 1968890 - Part 7: Add Cancel method. r=jonco
# ==============================================================================

diff -r 9e2a3327cd0f -r 02269004c755 dom/script/ScriptLoader.cpp
--- a/dom/script/ScriptLoader.cpp	Fri Sep 26 06:07:44 2025 +0000
+++ b/dom/script/ScriptLoader.cpp	Fri Sep 26 06:07:44 2025 +0000
@@ -4135,6 +4135,10 @@
   NS_ASSERTION(aRequest, "null request in stream complete handler");
   NS_ENSURE_TRUE(aRequest, NS_ERROR_FAILURE);
 
+  if (aRequest->IsCanceled()) {
+    return NS_BINDING_ABORTED;
+  }
+
   nsresult rv = VerifySRI(aRequest, aLoader, aSRIStatus, aSRIDataVerifier);
 
   if (NS_SUCCEEDED(rv)) {
@@ -4583,9 +4587,6 @@
     return aStatus;
   }
 
-  if (aRequest->IsCanceled()) {
-    return NS_BINDING_ABORTED;
-  }
   MOZ_ASSERT(aRequest->IsFetching());
   CollectScriptTelemetry(aRequest);
 